### PR TITLE
Add logic to display a premium indicator for themes in the showcase.

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* We talked about adding some kind of indicator for premium themes in #59966, even if the user already has the premium or business plans, to show the value they're getting.
* This has had no design or marketing input; I just wanted to get a handle on the logic. 

**The how**
* I check the `theme.price` prop and not the component `price` prop, as the latter is misleading; it changes depending on the user's access to these themes. So premium themes shown to a user on a premium plan have an empty `price` prop, but the `theme.price` on the object should still be greater than zero.
* Sometimes we get the price as a string and sometimes it's an object. For example, `theme.price` = `$50` on the Recommended tab, but `theme.price` = `{ value: 50, currency: 'USD' ... }` on the Trending or All Themes tabs. This adds some logic and could probably be simplified?

**Before**

Screen Shot 2022-01-18 at 12 36 07 PM

**After**

Screen Shot 2022-01-18 at 12 35 57 PM

#### Testing instructions

* Switch to this PR, navigate to `/themes`
* Check the Showcase on each plan type; Free, Personal, Premium, Business

The premium indicator should show for premium themes in the Showcase when:
* The user has a Premium or Business plan.
* The user has purchased an individual premium theme on a Personal or Free plan; the purchased theme(s) will show the indicator, but non-purchased themes will still show the price.
* The theme is not the active theme on the user's site.

### CODE REVIEW

File: client/components/theme/style.scss [hunk 1]
```diff
@@ -101,6 +101,7 @@ $theme-info-height: 54px;
 .theme__upsell-icon svg {
 	transform: scale( 0.8 );
 	border: 2px solid var( --color-neutral-20 );
+	/* stylelint-disable-next-line declaration-property-unit-allowed-list */
```
Comment (sixhours) [inline client/components/theme/style.scss:105]: Only added to appease the linter.

File: client/components/theme/style.scss [hunk 2]
```diff
@@ -124,7 +125,7 @@ $theme-info-height: 54px;
 
 .theme__badge-price-upgrade {
 	text-transform: uppercase;
-	font-size: 80%;
+	font-size: $font-body-small;
```
Comment (sixhours) [inline client/components/theme/style.scss:129]: Fixed to appease the linter/stay consistent with our use of font sizes.

**Comment by mreishus** [comment]
* :heavy_check_mark: 
Checked on premium simple, free simple, atomic and jetpack (to what extent is possible on jetpack) - all looks good, the badge is appearing when it should. 

* Suggestion: We are eventually trying to remove lodash ( p4TIVU-9Bf-p2 ), and it doesn't look like `isEmpty` is doing a lot for strings ([source](https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L11479)), so is replacing `! isEmpty( theme.price )` with `theme.price !== ""` okay?

* Also noticed a very slight alignment issue. This isn't a blocker and I don't know how to fix it, but I wanted to at least note it. Since both the theme name and the premium text are vertically centered, and different sizes, the baseline of the font isn't aligned:

It can be hard to notice, so by temporarily making the premium badge even smaller (0.5rem) the issue is exaggerated and more apparent:

**Comment by sixhours** [comment]
> Suggestion: We are eventually trying to remove lodash ( p4TIVU-9Bf-p2 ), and it doesn't look like isEmpty is doing a lot for strings (source), so is replacing ! isEmpty( theme.price ) with theme.price !== "" okay?

Ahh, good call, I think that's fine.

> Also noticed a very slight alignment issue. This isn't a blocker and I don't know how to fix it, but I wanted to at least note it. Since both the theme name and the premium text are vertically centered, and different sizes, the baseline of the font isn't aligned:

Yeah, I see what you mean. I'm not sure we'll actually use this as the design so I'll wait and polish it up once we have a better idea of what this should look like.

**Comment by sixhours** [comment]
👋 @SaxonF @ianstewart The logic here is working, do you have an idea of what you want the premium theme indicator to look like?

**Comment by kwight** [comment]
Business:

Screen Shot 2022-01-18 at 12 38 21 PM

Premium:

Screen Shot 2022-01-18 at 12 39 26 PM

Free:

Screen Shot 2022-01-18 at 12 40 10 PM

Looks good!

I do wonder about longer theme names and "Premium" translated to a more verbose language – we're working with limited space here. Does a top right corner ribbon help at all?..

**Comment by sixhours** [comment]
Ian noted the current design is "OK at best" so I'm going to polish up the spacing a bit and ship it. We can always revisit the design later and I don't want this to fall through the cracks. :)

**Comment by sixhours** [comment]
Screen Shot 2022-01-25 at 12 55 11 PM

Nudging the theme title up 1px seems to improve the vertical alignment across this row.

**Review by mreishus — APPROVED** [review]
LGTM :+1:
Premium plan:


Free plan, mix of purchased and unpurchased:

**Pull request merged**